packages/discovery-react-components/src/components/DocumentPreview/types.ts

```diff
@@ -78,7 +96,28 @@ export interface FacetInfo {
   displayName: string;
 }
 
-// Collection of facets
+// Collection of facets, map facet ID to facet info
 export type FacetInfoMap = Record<string, FacetInfo>;
 
+export interface OverlapInfo {
+  overlapId: string;
+  // Info about mentions included in the overlap.
+  mentions: DocumentFieldHighlight[];
+}
+
+// Collection of overlap meta data, map from overlap ID to overlap meta data
+export type OverlapInfoMap = Record<string, OverlapInfo>;
+
+export interface OverlapMeta {
+  overlapInfoMap: OverlapInfoMap;
+  fieldIdWithOverlap: Set<string>;
+}
+
+// OVERLAP_ID should be crafted to avoid conflicit with third party enrichment names
+export const OVERLAP_ID: string = 'OverlapFT';
+
+export function initOverlapMeta(): OverlapMeta {
```

Comment:
- added documentation.
- Still need function. The empty value will be assigned multiple times and all variables refer to the same object instead of their own instance. One variables' changes will be reflected for all other variables assigned to the empty object.

Reply:
Ah, good point!